Makefile

```diff
@@ -0,0 +1,35 @@
+
+#
+# This script installs the firmware files in the place where the linux
+# firmware loading system expects them - `/lib/firmware`. It supports
+# DESTDIR installations for package manager software. You should just
+# have to `sudo make install` to put the files in the correct location.
+#
+
+VERSION ?= v1.5.1
+
+FIRMWAREDIR ?= $(DESTDIR)/usr/lib/firmware
+
+all:
+
+check:
+
+install:
+	mkdir -p $(FIRMWAREDIR)
+	cp -rf lib/firmware/intel $(FIRMWAREDIR)/
+
+	mkdir -p $(FIRMWAREDIR)/sof
+
+	ln -s $(VERSION)/sof-bdw-$(VERSION).ri $(FIRMWAREDIR)/intel/sof/sof-bdw.ri
+	ln -s $(VERSION)/sof-byt-$(VERSION).ri $(FIRMWAREDIR)/intel/sof/sof-byt.ri
+	ln -s $(VERSION)/sof-cht-$(VERSION).ri $(FIRMWAREDIR)/intel/sof/sof-cht.ri
```

Comment:
Please:
```shell
 for p in bdw byt cht; do ln -s $(VERSION)/sof-$$p-$(VERSION).ri $(FIRMWAREDIR)/intel/sof/sof-$$p.ri
```

Reply:
in a Makefile?

Reply:
Each command in a Makefile is a shell script.

Reply:
BTW I don't see the point of making this a Makefile. A Makefile is a graph of connected commands/shell script with many entry points but so far this is a pure sequence and no graph.

BTW CMake has some installation features built-in but maybe they would be overkill for this. I'm not familiar with them, @jajanusz any clue?

Reply:
Auke suggested we used a Makefile coz that's what all distros use. And indeed for Debian the default is make install...

Reply:
`make install` makes sense as part of a bigger and actual makefile, hence my CMake question to @jajanusz . Purely embedding a shell script in a Makefile without using any Make dependency feature is just adding an extra layer of indirection/obfuscation. If we can't use CMake for some reason and if we really really need to implement a `make install` interface for compatibility with Debian/Fedora/etc. infrastructure then the best option is this:

```make
# dummy Makefile for compatibility reasons
.PHONY: install
install:
      go.sh
```

Reply:
talk to Auke. the patch comes from him, this works with Debian and ClearLinux, I don't see the point in debating

Reply:
As long as the size of this script disguised as a Makefile doesn't grow out of control it's fine. From that perspective embedding `for` loops helps.

Reply:
@marc-hb can you make the necessary changes for merging into master and re-submit. This will be quicker unless @plbossart objects ?

Reply:
no objections, too many things on my plate these days

Reply:
@plbossart thanks, will close this and wait for @marc-hb to create a new PR with his version.